src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -702,197 +647,136 @@ const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
-                            <PencilSimple className="h-6 w-6" />
-                          </Button>
-                          <TooltipElement id="rename-tooltip" delayShow={DELAY_SHOW_MS} />
-                        </div>
-                      )}
-                      <div
-                        className="flex items-center justify-center"
-                        data-tooltip-id="trash-tooltip"
-                        data-tooltip-content={translate('drive.dropdown.moveToTrash')}
-                        data-tooltip-place="bottom"
-                      >
-                        <Button variant="tertiary" className="aspect-square" onClick={onBulkDeleteButtonClicked}>
-                          <Trash className="h-6 w-6" />
-                        </Button>
-                        <TooltipElement id="trash-tooltip" delayShow={DELAY_SHOW_MS} />
-                      </div>
-                    </div>
-                  </>
-                )}
-                {separatorV}
-                <div className="flex items-center justify-center">
-                  <Button variant="tertiary" className="aspect-square" onClick={onViewModeButtonClicked}>
-                    {viewModesIcons[viewMode]}
-                  </Button>
-                  <TooltipElement id="viewMode-tooltip" delayShow={DELAY_SHOW_MS} />
-                </div>
-              </div>
-            )}
+                                    )}
+                                  </Menu.Item>
+                                  <Menu.Item>
+                                    {({ active }) => (
+                                      <div
+                                        onClick={onUploadFolderButtonClicked}
+                                        className={`${
+                                          active && 'bg-gray-5'
+                                        } flex cursor-pointer items-center space-x-3 whitespace-nowrap py-2 pl-3 pr-5 text-gray-80 hover:bg-gray-5`}
+                                      >
+                                        <UploadSimple size={20} />
+                                        <p className="ml-3">{translate('actions.upload.uploadFolder')}</p>
+                                      </div>
+                                    )}
+                                  </Menu.Item>
+                                </Menu.Items>
+                              )}
+                            </Transition>
+                          </>
+                        );
+                      }}
+                    </Menu>
+                  </div>
+                  {!isTrash && <DriveTopBarItems />}
+                </>
+              )}
+              <DriveTopBarActions
+                hasAnyItemSelected={hasAnyItemSelected}
+                isTrash={isTrash}
+                selectedItems={selectedItems}
+                separatorV={separatorV}
+                tooltipDelay={DELAY_SHOW_MS}
```

Comment:
No need to pass this as props, the DELAY_SHOW_MS can be imported into the component. And the separator could be left here or in the new component, it doesn't need to be in both as far as I've seen.